src/attacks.rs

```diff
@@ -14,15 +22,214 @@ pub enum DepthReduceSet {
     /// Parameter  is size of the resulting G-S graph desired
     ValiantAB16(usize),
     /// depth of the resulting G-S graph and some specific parameters
-    Greedy(usize, GreedyParams),
+    GreedyDepth(usize, GreedyParams),
+    /// Variation of Greedy attack that has as `target` the resulting size of set S.
+    GreedySize(usize, GreedyParams),
 }
 
 pub fn depth_reduce(g: &mut Graph, drs: DepthReduceSet) -> HashSet<usize> {
     match drs {
         DepthReduceSet::ValiantDepth(_) => valiant_reduce(g, drs),
         DepthReduceSet::ValiantSize(_) => valiant_reduce(g, drs),
         DepthReduceSet::ValiantAB16(_) => valiant_reduce(g, drs),
-        DepthReduceSet::Greedy(target, p) => greedy_reduce(g, target, p),
+        DepthReduceSet::GreedyDepth(_, _) => greedy_reduce(g, drs),
+        DepthReduceSet::GreedySize(_, _) => greedy_reduce(g, drs),
+    }
+}
+
+/// Target of an attack, either the depth should be smaller than `Depth` or
+/// the exclusion set `S` should have a size bigger than `Size` (actually
+/// we want to hit a target as close as possible as those thresholds). All
+/// targets are specified in relation to the size of the graph `G` (not to
+/// be confused with the target size of set `S`).
+// FIXME: Overlapping with `DepthReduceSet` internal values, this should
+//  replace that.
+#[derive(Debug)]
+pub enum AttackTarget {
+    Depth(f64),
+    Size(f64),
+}
+
+/// Range of targets to try (to find the optimum value) from `start`, increasing
+/// by `interval` until `end` is reached or surpassed.
+// FIXME: Using this instead of `std::ops::Range<f64>` because Rust correctly
+//  doesn't allow iterating over floating point values but there is probably
+//  an easier way than coding this from scratch.
+// FIXME: Assert range validity in the struct itself instead of on the caller
+//  (`attack`).
+#[derive(Debug)]
+pub struct TargetRange {
```

Comment:
Just to say that this is probably overkill for what we need and an array of values should be sufficient to pass on and may simplify some logic in the rest of PR as well. 
That being said, if it's all good on your side, then go for it :)